purchases/index.js

```diff
@@ -528,28 +529,50 @@ function listPurchases(db, options, callback) {
     });
 }
 
+function createWhereStatement(options) {
+    const parts = [parseCategory(options), parseAfter(options), parseBefore(options)].filter(x => x !== "");
+    if(parts.length === 0) {
+        return "";
+    }
+    return `WHERE ${parts.join(" AND ")}`;
+}
+
 function parseCategory({categories}) {
     if(!categories) {
         return "";
     }
     categoriesAsArray = categories.split(",");
-    return `WHERE status IN (${categoriesAsArray.map(category => `'${category}'`).join(", ")})`;   
+    return `status IN (${categoriesAsArray.map(category => `'${category}'`).join(", ")})`;   
+}
+
+function parseAfter({after}) {
+    if(!after) {
+        return "";
+    }
+    return `(data ->> 'completedAt')::timestamptz > '${after}'::timestamptz`;
+}
+
+function parseBefore({before}) {
+    if(!before) {
+        return "";
+    }
+    return `(data ->> 'completedAt')::timestamptz <= '${before}'::timestamptz`;
```

Comment:
Have you tested what happens if given an invalid value for `before` or `after`? Do we handle that gracefully and give a good error message?

Reply:
i have now set some basic validation, so it only filters on dates if a valid date is provided.